test/jest/cookieSyncManager.spec.ts

```diff
@@ -0,0 +1,454 @@
+                false
+            );
+
+            // Simulate image load
+            mockImage.onload();
+
+            expect(mockImage.src).toBe('https://test.com');
+            expect(cookieSyncDates[42]).toBeDefined();
+            expect(cookieSyncDates[42]).toBeGreaterThan(0);
+        });
+
+        it('should log a verbose message when a cookie sync is performed', () => {
+            const mockImage = {
+                onload: jest.fn(),
+                src: '',
+            };
+            jest.spyOn(document, 'createElement').mockReturnValue(
+                (mockImage as unknown) as HTMLImageElement
+            );
+
+            const loggerSpy = jest.fn();
+
+            const mockSDK = ({
+                _Store: {
+                    webviewBridgeEnabled: false,
+                    pixelConfigurations: [pixelSettings],
+                },
+                _Persistence: {
+                    setCookieSyncDates: jest.fn(),
+                    getPersistence: jest.fn(),
+                    saveUserCookieSyncDatesToPersistence: jest.fn(),
+                },
+                _Consent: {
+                    isEnabledForUserConsent: jest.fn().mockReturnValue(true),
+                },
+                Identity: {
+                    getCurrentUser: jest.fn().mockReturnValue({
+                        getMPID: () => '123',
+                    }),
+                },
+                Logger: {
+                    verbose: loggerSpy,
+                },
+            } as unknown) as MParticleWebSDK;
+
+            const cookieSyncManager = new CookieSyncManager(mockSDK);
+
+            let cookieSyncDates = [];
+            cookieSyncManager.performCookieSync(
+                'https://test.com',
+                42,
+                '1234',
+                cookieSyncDates,
+                null,
+                false,
+                false
+            );
+
+            // Simulate image load
+            mockImage.onload();
+
+            expect(loggerSpy).toHaveBeenCalledWith('Performing cookie sync');
+        });
+
+        it('should return early if the user has not consented to the cookie sync', () => {
+            const mockImage = {
+                onload: jest.fn(),
+                src: '',
+            };
+            jest.spyOn(document, 'createElement').mockReturnValue(
+                (mockImage as unknown) as HTMLImageElement
+            );
+
+            const loggerSpy = jest.fn();
+
+            const mockSDK = ({
+                _Store: {
+                    webviewBridgeEnabled: false,
+                    pixelConfigurations: [pixelSettings],
+                },
+                _Persistence: {
+                    setCookieSyncDates: jest.fn(),
+                    getPersistence: jest.fn(),
+                    saveUserCookieSyncDatesToPersistence: jest.fn(),
+                },
+                _Consent: {
+                    isEnabledForUserConsent: jest.fn().mockReturnValue(true),
```

Comment:
should this be false?  isEnabled being true makes me think that the cookie sync would happen, and so the value for cookieSyncDates[42] should be defined.  What am i missing here?